src/zenml/steps/external_artifact.py

```diff
@@ -138,6 +153,17 @@ def upload_if_necessary(self) -> UUID:
                     "issues loading the artifact. Please make sure to only "
                     "reference artifacts stored in your active artifact store."
                 )
+        elif self._pipeline_name and self._artifact_name:
```

Comment:
I think this would benefit from the same check as the one with `self._id` above, making sure that the artifact is located in the same artifact store as the one being used by the current pipeline run.

Reply:
Other than that, love the idea of passing pipeline/artifact name! We could even add a small reference to that in the docs here: https://docs.zenml.io/user-guide/advanced-guide/pipelining-features/configure-steps-pipelines#pass-any-kind-of-data-to-your-steps

Reply:
Thanks @schustmi  - I reworked the logic to incorporate that, can you have a second look pls?

I'm also unsure how to update docs - will reach out by other channels for guidance 🙂